werkit/compute/tree/sketch.py

```diff
@@ -0,0 +1,43 @@
+"""
+References:
+
+https://github.com/ivankorobkov/python-inject
+https://github.com/illuin-tech/opyoid
+https://github.com/ets-labs/python-dependency-injector
+"""
+
+
+class MyComputeProcess:
+    a = Input(_type="number")
+    b = Input(_type="number")
+
+    @intermediate(_type="number")
+    def i(self, a):
+        return a
+
+    @intermediate(_type="number")
+    def j(self, b):
+        return b
+
+    @output(_type="number")
+    def r(self, i, j):
+        return i + j
+
+
+compute_process = MyComputeProcess()
+
+# Inject dependencies.
+inject(compute_process, a=1, b=2)
```

Comment:
Is the idea here that the compute process should support partial evaluation of MyComputeProcess? 

Because I don't know if we need it to do that, as this will be handled by the compute tree.

It seems like, instead, all we need are pure functions (and input variables) that reference other pure functions (and input variables) as dependencies. One difference between that approach and what is presented here is that this `inject` call seems to set the state of the compute process. I don't think we will need that, because the state will be stored in the compute graph on the client. Instead, what we would need is to simply call the pure function with the arguments it expects to receive. 

You also might not want to group the pure functions into a single compute process class, as this could make it harder to compose the pure functions into multiple compute processes (multiple "templates").